tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/domain/ExerciseBuilder.java

```diff
@@ -10,17 +10,20 @@
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class ExerciseBuilder {
 
-    private static final String BEGIN_SOLUTION_TAG = "// BEGIN SOLUTION";
-    private static final String END_SOLUTION_TAG = "// END SOLUTION";
-    private static final String STUB_TAG = "// STUB:";
-    private static final String SOLUTION_FILE_TAG = "// SOLUTION FILE";
+    private static final String BEGIN_SOLUTION_REGEX = ".*\\/\\/[ \\t]*BEGIN[ \\t]+SOLUTION.*";
+    private static final String END_SOLUTION_REGEX = ".*\\/\\/[ \\t]*END[ \\t]+SOLUTION.*";
+    private static final String SOLUTION_FILE_REGEX = ".*\\/\\/[ \\t]*SOLUTION[ \\t]+FILE.*";
+    private static final String STUB_REGEX = "(.*)\\/\\/[ \\t]*STUB:[ \\t]*(.*)";
```

Comment:
I'm not sure if we want to have the `.*` in the beginning, maybe we want just to allow having whitespace before these tags?